### PR TITLE
implement Storage by HashMap

### PR DESCRIPTION
**Pull request by fed4ik**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -1,19 +1,32 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.HashMap;
+import java.util.Map;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private Map<K,V> map;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:8]: image

**Review by Rommelua — CHANGES_REQUESTED** [review]
https://mate-academy.github.io/jv-program-common-mistakes/java-core/generics/storage.html

Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -1,19 +1,51 @@
...
     public void put(K key, V value) {
+        if (putItemsCopy(key, value)) {
+            return;
+        }
+        keyArray[itemsNumber] = key;
+        valueArray[itemsNumber] = value;
+        itemsNumber++;
+    }
+
+    private boolean putItemsCopy(K key, V value) {
```
Comment (andriy-shulepa) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:27]: private methods should be below private ones

**Review by andriy-shulepa — CHANGES_REQUESTED** [review]
https://mate-academy.github.io/jv-program-common-mistakes/java-core/generics/storage.html

Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

image

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (fed4ik) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Could you rephrase last sentence or maybe give advise where I can read about it?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private final K[] keyArray;
+    private final V[] valueArray;
+    private int itemsNumber = 0;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]:
```suggestion
    private int size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: below public ones)

**Review by Rommelua — APPROVED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (fed4ik) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: )))) even the chatGPT was confused!)
